Model/CustomItemExportSchedulerModel.php

```diff
@@ -0,0 +1,294 @@
+use MauticPlugin\CustomObjectsBundle\Repository\CustomItemRepository;
+use MauticPlugin\CustomObjectsBundle\Repository\CustomItemXrefContactRepository;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+class CustomItemExportSchedulerModel extends AbstractCommonModel
+{
+    private const EXPORT_FILE_NAME_DATE_FORMAT = 'Y_m_d_H_i_s';
+    private const CUSTOM_ITEM_LIMIT            = 200;
+    private const CONTACT_LIMIT                = 5000;
+
+    private ExportHelper $exportHelper;
+
+    private MailHelper $mailHelper;
+
+    private CustomItemRouteProvider $customItemRouteProvider;
+
+    private CustomFieldValueModel $customFieldValueModel;
+
+    private CustomItemXrefContactRepository $customItemXrefContactRepository;
+
+    private CustomItemRepository $customItemRepository;
+
+    private EventDispatcherInterface $eventDispatcher;
+
+    private string $filePath;
+
+    public function __construct(
+        ExportHelper $exportHelper,
+        MailHelper $mailHelper,
+        CustomFieldValueModel $customFieldValueModel,
+        CustomItemRouteProvider $customItemRouteProvider,
+        CustomItemXrefContactRepository $customItemXrefContactRepository,
+        CustomItemRepository $customItemRepository,
+        EventDispatcherInterface $eventDispatcher
+    ) {
+        $this->exportHelper                    = $exportHelper;
+        $this->mailHelper                      = $mailHelper;
+        $this->customFieldValueModel           = $customFieldValueModel;
+        $this->customItemRouteProvider         = $customItemRouteProvider;
+        $this->customItemXrefContactRepository = $customItemXrefContactRepository;
+        $this->customItemRepository            = $customItemRepository;
+        $this->eventDispatcher                 = $eventDispatcher;
+    }
+
+    public function getRepository(): CustomItemExportSchedulerRepository
+    {
+        /** @var CustomItemExportSchedulerRepository $repo */
+        $repo = $this->em->getRepository(CustomItemExportScheduler::class);
+
+        return $repo;
+    }
+
+    /**
+     * @throws \Doctrine\ORM\ORMException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     */
+    public function saveEntity(int $customObjectId): CustomItemExportScheduler
+    {
+        $customItemExportScheduler = new CustomItemExportScheduler();
+        $customItemExportScheduler
+            ->setUser($this->userHelper->getUser())
+            ->setScheduledDateTime(new DateTimeImmutable('now', new DateTimeZone('UTC')))
+            ->setCustomObjectId($customObjectId);
+
+        $this->em->persist($customItemExportScheduler);
+        $this->em->flush();
+
+        return $customItemExportScheduler;
+    }
+
+    public function processDataAndGetExportFilePath(CustomItemExportScheduler $customItemExportScheduler): string
```

Comment:
I dont think the model is the right place for processing and exporting the data. This should be extracted into its own class/service IMO